test/test_corpus.cc

```diff
@@ -601,6 +602,110 @@ TEST_P(LSPTest, All) {
         }
     }
 
+    // Quick fix code action
+    vector<shared_ptr<ApplyCodeActionAssertion>> applyCodeActionAssertions;
+    for (auto &assertion : assertions) {
+        if (auto applyCodeActionAssertion = dynamic_pointer_cast<ApplyCodeActionAssertion>(assertion)) {
+            applyCodeActionAssertions.push_back(applyCodeActionAssertion);
+        }
+    }
+    bool exhaustiveApplyCodeAction =
+        BooleanPropertyAssertion::getValue("exhaustive-apply-code-action", assertions).value_or(false);
+
+    if (!applyCodeActionAssertions.empty() || exhaustiveApplyCodeAction) {
+        // TODO(sushain): test multi-file code actions
+        ASSERT_EQ(filenames.size(), 1) << "code-actions only works with tests that have a single file.";
+
+        auto errors = RangeAssertion::getErrorAssertions(assertions);
+
+        // Request code actions for each error.
+        for (auto &error : errors) {
+            vector<unique_ptr<Diagnostic>> diagnostics;
+            auto fileUri = testFileUris[*filenames.begin()];
+            // TODO(sushain): stop manually copying these over
+            auto params = make_unique<TextDocumentCodeActionParams>(
+                make_unique<TextDocumentIdentifier>(fileUri),
+                make_unique<Range>(make_unique<Position>(error->range->start->line, error->range->start->character),
+                                   make_unique<Position>(error->range->end->line, error->range->end->character)),
+                make_unique<CodeActionContext>(move(diagnostics)));
+            auto req = make_unique<RequestMessage>("2.0", nextId++, LSPMethod::TextDocumentCodeAction, move(params));
+            auto responses = lspWrapper->getLSPResponsesFor(LSPMessage(move(req)));
+            EXPECT_EQ(responses.size(), 1) << "Did not receive a response for a codeAction request.";
```

Comment:
... or received multiple responses. ("Did not receive expected response for a codeAction request." ?)